lib/matching/nearest_neighbour.h

```diff
@@ -11,14 +11,20 @@
 
 namespace MouseTrack {
 
-class NearestNeighbour: public Matching {
-	public:
-		/// Given a list of cluster chains, we find the best matching to append the given list of descriptors.
-		/// descriptors: a list of descriptors from a new frame
-		/// chains: a list of cluster chains to which we want to appeand one cluster from descriptors
-		/// returns a vector of size descriptors.size(), which holds the index of the matching chain for each descriptor.
-		/// If no match was found, the vector value is set to -1.
-		virtual std::vector<long> operator()(const std::vector<std::shared_ptr<const ClusterDescriptor>>& descriptors, const std::vector<ClusterChain>& chains);
+class NearestNeighbour : public Matching {
+
+public:
+  virtual ~NearestNeighbour() {}
+
+  /// Given a list of cluster chains, we find the best matching to append the
+  /// given list of descriptors. descriptors: a list of descriptors from a new
```

Comment:
Automatic formatting messed with newlines, I think, I need to place two line-breaks in the future.

Reply:
What do you mean? I like it this way. Either way, the LLVM styler automatically cuts two lines to one line.

Reply:
See line 21: "frame chains:" frame belongs to the previous line, it was merged with the next line which makes it really hard to read. Same thing line 20: "... descriptors. descriptors: " and line 22 "cluster from descriptors returns a vector"

Reply:
I understand now. Fixed this case, will try to fix them as I see them